test/call.c

```diff
@@ -3815,3 +3816,101 @@ int test_call_cert_select(void)
 	return err;
 }
 #endif
+
+
+static void sip_server_exit_handler(void *arg)
+{
+	(void)arg;
+	re_cancel();
+}
+
+
+int test_call_uag_find_msg(void)
+{
+	struct fixture fix, *f = &fix;
+	struct sip_server *srv1 = NULL;
+	struct sip_server *srv2 = NULL;
+	struct sa sa1;
+	struct sa sa2;
+	char *aor=NULL;
+	struct cancel_rule *cr;
+	int err = 0;
+
+	fixture_init(f);
+
+	err = sip_server_alloc(&srv1, sip_server_exit_handler, NULL);
+	TEST_ERR(err);
+
+	err = sip_server_alloc(&srv2, sip_server_exit_handler, NULL);
+	TEST_ERR(err);
+
+	err = sip_transp_laddr(srv1->sip, &sa1, SIP_TRANSP_UDP, NULL);
+	TEST_ERR(err);
+
+	err = sip_transp_laddr(srv2->sip, &sa2, SIP_TRANSP_UDP, NULL);
+	TEST_ERR(err);
+
+	mem_deref(f->a.ua);
+	mem_deref(f->b.ua);
+	mem_deref(f->c.ua);
+
+	err = re_sdprintf(&aor, "A <sip:alice@%J>;regint=60", &sa1);
+	TEST_ERR(err);
+	err = ua_alloc(&f->a.ua, aor);
+	TEST_ERR(err);
+	mem_deref(aor);
+	err = re_sdprintf(&aor, "B <sip:alice@%J>;regint=60", &sa2);
+	TEST_ERR(err);
+	err = ua_alloc(&f->b.ua, aor);
+	TEST_ERR(err);
+	mem_deref(aor);
+	err = re_sdprintf(&aor, "C <sip:bob@%J>;regint=60", &sa2);
+	TEST_ERR(err);
+	err = ua_alloc(&f->c.ua, aor);
+	TEST_ERR(err);
+
+	err = ua_register(f->a.ua);
+	TEST_ERR(err);
+	err = ua_register(f->b.ua);
+	TEST_ERR(err);
+	err = ua_register(f->c.ua);
```

Comment:
account C is used for outgoing call, is it required to register it ?

Reply:
a SIP account does not have to be registered in order to make outgoing call...

The selection of transport option for INVITE/200-OK/ACK does not depend on
if the endpoint is registered or not.